OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -371,49 +293,25 @@ ActorInfo ChooseUnitToBuild(ProductionQueue queue)
 			return null;
 		}
 
-		IEnumerable<Actor> GetActorsWithTrait<T>()
+		bool HasAdequateConstructionYardCount
```

Comment:
These ones return consistent results (assuming other state doesn't change), so are appropriate to leave as properties.